social/backends/saml.py

```diff
@@ -0,0 +1,289 @@
+"""
+Backend for SAML 2.0 support
+
+Terminology:
+
+"Service Provider" (SP): Your web app
+"Identity Provider" (IdP): The third-party site that is authenticating users via SAML
+"""
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
+from social.backends.base import BaseAuth
+from social.exceptions import AuthFailed
+
+# Helpful constants:
+OID_COMMON_NAME = "urn:oid:2.5.4.3"
+OID_EDU_PERSON_PRINCIPAL_NAME = "urn:oid:1.3.6.1.4.1.5923.1.1.1.6"
```

Comment:
I think what I am confused about is that "urn:oid:1.3.6.1.4.1.5923.1.1.1.6" for example. I may be naive but isn't `urn:oid:1.3.6.1.4.1.5923.1.1.1.6` defined specifically as an attribute for Principle Name for the purposes of University/Higher Education LDAP systems? If it was an LDAP for a corporation, it would use a different attribute?